runtime/near-test-contracts/congestion-control-test-contract/src/lib.rs

```diff
@@ -76,6 +76,8 @@ extern "C" {
     // #######################
     fn promise_batch_action_create_account(promise_index: u64);
     fn promise_batch_action_deploy_contract(promise_index: u64, code_len: u64, code_ptr: u64);
+    fn promise_batch_action_deploy_global_contract(promise_index: u64, code_len: u64, code_ptr: u64);
+    fn promise_batch_action_deploy_global_contract_by_account_id(promise_index: u64, code_len: u64, code_ptr: u64);
```

Comment:
Huh, why is deploying a global contract by an account id taking a full code to deploy and not the account id to reference?

Reply:
when deploying global contract by account id we use account id of the signer (receipt's `predecessor_id`), also see `DeployGlobalContractAction`

Reply:
@pugachAG Is it possible to name the pointers properly, so the naming is not misleading?

Reply:
@pugachAG Do you mean current_account_id where the contract is executed at? It is totally odd to deploy the contract to the signer/predecessor account!!!

Will it actually replace the currently deployed contract on the account_id?

Would you be open to naming it `promise_batch_action_deploy_global_contract_identified_by_account_id`?

Reply:
> Do you mean current_account_id where the contract is executed at? 

Technically this corresponds to the receipt receiver account_id. `DeployGlobalContract` action requires `predecessor_id` to be equal to the `receiver_id`, so those should be the same.

> Will it actually replace the currently deployed contract on the account_id?

No, deploying global contract does not replace anything, it just makes the contract available. `UseGlobalContract` does that.

> Would you be open to naming it 

I prefer to keep host function naming consistent with action naming. This name is derived from `ActionView::DeployGlobalContractByAccountId`